IanGRap/Squish
# Treasure grab ui

Contributor: Coulter

Goal: Add UI above player heads to track current number of points over total points needed to win the round.

Files Changed:
- Player.prefab: Added score text to the player canvas that appears above the player's head and tracks the number of points they have at any given time.
- TreasureUI.cs: Small script that changes the above UI text as appropriate.
- TreasureGrabModeManager.cs: Made into a singleton so that TreasureUI.cs could access the PointsRequired property.
- Treasure Grab Test.unity: Changed required win and point values to what is currently expected to be the default for the game mode.

How to test:
- Run the Treasure Grab Test.unity scene.
- Play through the game (break treasure blocks, kill players, etc.) to make sure that the score values above player heads are correct.
- Change required win and point values in [Treasure Grab Match Manager].prefab and then play again to ensure all UI (text above player heads and UI at end of match) still shows correct values.
- Make sure no UI weirdness was introduced into the other modes. Shouldn't be an issue, but still safe to check.

Notes: The exact positioning and size of the new UI can be changed pretty easily. Making it more like it was in the prototype build, where the new UI for all players was always at the top of the screen, would be harder, especially since even in the prototype there were bugs with that UI. I think we should stick with UI over player heads unless playtest feedback says to do otherwise.
